ecl/eclagent/eclgraph.cpp

```diff
@@ -1811,9 +1811,13 @@ void EclAgent::executeGraph(const char * graphName, bool realThor, size32_t pare
             }
 
             activeGraph.setown(loadGraph(graphName, queryWorkUnit(), dll, false));
-            unsigned guillotineCost = queryWorkUnit()->getDebugValueInt("maxCost", 0);
+            unsigned guillotineCost = queryWorkUnit()->getDebugValueInt("maxCost", softCostLimit);
             if (guillotineCost)
+            {
+                if (hardCostLimit && guillotineCost > hardCostLimit)
```

Comment:
I think it's ok, but had to check operator precedence, it would clearer if you added brackets, i.e.
```
if (hardCostLimit && (guillotineCost > hardCostLimit))
```